ARCs/arc-0004.md

```diff
@@ -555,42 +588,40 @@ group, which is undesirable. Instead, the locations of the
 transactions are implied entirely by the placement of the transaction
 types in the argument list.
 
-For example, to invoke the method `deposit(string,axfer,uint32)void`,
-a client would create a transaction group containing an asset transfer
-followed by the application call.  When encoding the other
-(non-transaction) arguments, the client should act as if the
-transaction arguments were completely absent from the method
-signature.  The application call would contain the method selector in
-ApplicationArgs[0], the first (string) argument in ApplicationArgs[1],
-and the third (uint32) argument in ApplicationArgs[2].
-
-> Further questions: Should it be possible to describe requirements of
-> certain fields of the transactions named this way? For example, can
-> the amount of recipient of a pay transaction be specified? A
-> specific, important instance of this problem: How can logicsigs on
-> these transactions be provided in the JSON interface?
+For example, to invoke the method `deposit(string,axfer,pay,uint32)void`,
+a client would create a transaction group containing, in this order:
+1. an asset transfer
+2. a payment
+3. the actual application call
+When encoding the other (non-transaction) arguments, the client 
+**MUST** act as if the transaction arguments were completely absent 
+from the method signature. The application call would contain the method 
+selector in ApplicationArgs[0], the first (string) argument in 
+ApplicationArgs[1], and the fourth (uint32) argument in ApplicationArgs[2].
 
 
 ARC-4 Apps **SHOULD** be constructed to allow their invocations to be
 combined with other contract invocations in a single atomic group if
-they can do so safely. For example, they should use `txns` to examine
+they can do so safely. For example, they **SHOULD** use `txns` to examine
 the previous index in the group for a required `pay` transaction,
 rather than hardcode an index with `txn`.
```

Comment:
```suggestion
they can do so safely. For example, they **SHOULD** use `gtxns` to examine
the previous index in the group for a required `pay` transaction,
rather than hardcode an index with `gtxn`.
```

Tangential to this change, but this should be `gtxns` and `gtxn`